adam_core/backend/pyoorb.py

```diff
@@ -0,0 +1,326 @@
+                vy : heliocentric ecliptic J2000 y velocity in AU per day
+                vz : heliocentric ecliptic J2000 z velocity in AU per day
+            If keplerian:
+                a : semi-major axis in AU
+                e : eccentricity in degrees
+                i : inclination in degrees
+                Omega : longitude of the ascending node in degrees
+                omega : argument of periapsis in degrees
+                M0 : mean anomaly in degrees
+            If cometary:
+                p : perihelion distance in AU
+                e : eccentricity in degrees
+                i : inclination in degrees
+                Omega : longitude of the ascending node in degrees
+                omega : argument of periapsis in degrees
+                T0 : time of perihelion passage in MJD
+        time_scale : {'UTC', 'UT1', 'TT', 'TAI'}, optional
+            Time scale of the MJD epochs.
+        magnitude : float or `~numpy.ndarray` (N), optional
+            Absolute H-magnitude or M1 magnitude.
+        slope : float or `~numpy.ndarray` (N), optional
+            Photometric slope parameter G or K1.
+
+        Returns
+        -------
+        orbits_pyoorb : `~numpy.ndarray` (N, 12)
+            Orbits formatted in the format expected by PYOORB.
+                orbit_id : index of input orbits
+                elements x6: orbital elements of propagated orbits
+                orbit_type : orbit type
+                epoch_mjd : epoch of the propagate orbit
+                time_scale : time scale of output epochs
+                H/M1 : absolute magnitude
+                G/K1 : photometric slope parameter
+        """
+        orbits_ = orbits.copy()
+        if orbits_.shape == (6,):
+            num_orbits = 1
+        else:
+            num_orbits = orbits_.shape[0]
+
+        if orbit_type == "cartesian":
+            orbit_type_list = [1 for i in range(num_orbits)]
+        elif orbit_type == "cometary":
+            orbit_type_list = [2 for i in range(num_orbits)]
+            # H = M1
+            # G = K1
+            orbits_[:, 1:5] = np.radians(orbits_[:, 1:5])
+        elif orbit_type == "keplerian":
+            orbit_type_list = [3 for i in range(num_orbits)]
+            orbits_[:, 1:] = np.radians(orbits_[:, 1:])
+        else:
+            raise ValueError(
+                "orbit_type should be one of {'cartesian', 'keplerian', 'cometary'}"
+            )
+
+        if time_scale == "UTC":
+            time_scale_list = [1 for i in range(num_orbits)]
+        elif time_scale == "UT1":
+            time_scale_list = [2 for i in range(num_orbits)]
+        elif time_scale == "TT":
+            time_scale_list = [3 for i in range(num_orbits)]
+        elif time_scale == "TAI":
+            time_scale_list = [4 for i in range(num_orbits)]
+        else:
+            raise ValueError("time_scale should be one of {'UTC', 'UT1', 'TT', 'TAI'}")
```

Comment:
I think it's nice to use [`enum`](https://docs.python.org/3/library/enum.html) for things like this. You'd get type checking and the code will end up being a lot less verbose. 

```python
class OpenOrbTimescale(enum.Enum):
    UTC = 1
    UT1 = 2
    TT = 3
    TAI = 4

...

def _configure_orbits(
        orbits: np.ndarray,
        t0: np.ndarray,
        orbit_type: str,
        time_scale: OpenOrbTimescale,
        magnitude: Optional[float] = None,
        slope: Optional[float] = None,
    ) -> np.ndarray:

        ...
        time_scale_list = [time_scale.value for i in range(num_orbits)]   
```

Called like `PYOORB._configure_orbits(..., time_scale=OpenOrbTimescale.UTC, ...)`

Reply:
I've added two enums (one for orbit types and the other for time scales).